cylc/flow/data_store_mgr.py

```diff
@@ -651,34 +651,54 @@ def generate_definition_elements(self):
         self.parents = parents
 
     def increment_graph_window(
-            self, itask, edge_distance=0, active_id=None,
-            descendant=False, is_parent=False):
-        """Generate graph window about given origin to n-edge-distance.
+            self,
+            source_tokens,
+            point,
+            flow_nums,
+            edge_distance=0,
+            active_id=None,
+            descendant=False,
+            is_parent=False,
+            is_manual_submit=False,
+            itask=None
+    ):
+        """Generate graph window about active task proxy to n-edge-distance.
+
+        A recursive function, that creates a node then moves to children and
+        parents repeating this process out to one edge beyond the max window
+        size (in edges). Going out one edge further, we can trigger
+        pruning as new active tasks appear beyond this boundary.
+
 
         Args:
-            itask (cylc.flow.task_proxy.TaskProxy):
-                Update task-node from corresponding task proxy object.
+            source_tokens (cylc.flow.id.Tokens)
+            point (PointBase)
+            flow_nums (set)
             edge_distance (int):
                 Graph distance from active/origin node.
             active_id (str):
                 Active/origin node id.
```

Comment:
Sorry, I'm still a little confused as to the difference between `source_tokens` and `active_id`

Reply:
`active_id` is the ID or the active task proxy (the origin about which the graph is expanded), it is passed through the recursive calls to `increment_graph_window` by it's children/parents.

The purpose is to keep an association of that task's window nodes/branch (children/parents), so we know what to prune (or check for pruning) when that active task leaves the pool.

Reply:
`source_tokens` will change throughout the recursion; from the active pool task's (`itask.tokens`) to the child/parent tokens generated in the graph walk.